stairway/src/main/java/bio/terra/stairway/FlightMap.java

```diff
@@ -120,6 +133,38 @@ boolean isEmpty() {
     }
   }
 
+  /**
+   * Return the object from the hash map deserialized to the right type. Throw an exception if the
+   * Object cannot be deserialized to that type. This overload is preferred when deserializing
+   * parameterized types as it provides stronger type checking at deserialization time.
+   *
+   * @param <T> - type of class to expect in the hash map
+   * @param key - key to lookup in the hash map
+   * @param typeReference - class requested
```

Comment:
It's a shame the api is fixed as this is great case for `Optional`

Reply:
Given this is a new call we could make this `Optional` if we don't care about mixing return types on overloads.  @ddietterich what do you think?

Reply:
Does Java let you change the return type of an overloaded method? Even if it does, I think that would be unexpected for users if the api. Instead I think you'd have to deprecate `get` and add a new method.